File: app/models/is_assertions.rb
module IsAssertions

  def assert(pass, msg="Assert Failed", skip=1)
    raise StandardError, msg, caller(skip) if !pass
  end

  def is(value, kind)
    assert(value.is_a?(kind), "#{value} must be a kind of #{kind}", 2)
    value
  end

end

File: js/client-image/is-assertions.js
var is = function (value, type) {
  if ((typeof value !== type) && !(value instanceof type)) {
    throw "value which is " + value + " must be of type " + type;
  }
}

is.assert = function (pass, msg) {
  if (!pass) {
    throw msg;
  }
}

module.exports = is;